src/main/kotlin/notification/domain/NotificationRepository.kt

```diff
@@ -1,8 +1,10 @@
 package notification.domain
 
+import deadline.domain.Time
 import shared.domain.Identifier
 
 interface NotificationRepository {
     fun search(id: Identifier): Collection<Notification>
     fun create(notification: Notification)
+    fun remove(notification: Notification)
```

Comment:
Maybe removing it via the identifier makes more sense?

Reply:
Notifications don't have an id, they have a note identifier related to them but both the note id and the time are needed to tell which notification is the one to be deleted since there can be more than one notification for the same note id.